### PR TITLE
Feat/barchart

### PR DESCRIPTION
**Pull request by persas**

### CODE REVIEW

**Review by SteRiccio — COMMENTED** [review]
I would also add internationalization to all the blocks' titles and subtitles

**Comment by SteRiccio** [comment]
also please check the SonarCloud findings, it seems that there are some duplicate lines

File: webapp/views/App/views/Data/Charts/state/chartTypes/pie/index.js [hunk 1]
```diff
@@ -29,88 +19,164 @@ const pie = {
...
-                    ...spec,
-                  }
-                  delete newSpec.encoding.color
-                  onUpdateSpec(A.stringify(newSpec))
-                  return
-                }
+          groupBy: GroupByBlock({
+            valuesToSpec: ({ value = [], spec = {} }) => {
+              const transform = {
+                calculate: `${value.map((val) => `datum.${val.value}`).join("+','+")}`,
```
Comment (SteRiccio) [inline webapp/views/App/views/Data/Charts/state/chartTypes/pie/index.js:25]: the way to determine the "calculate" is repeated several times. Can you please extract it into some helper function?

File: webapp/views/App/views/Data/Charts/state/hooks/useChartConfig.js [hunk 2]
```diff
@@ -34,6 +45,8 @@ const updateItemByPathAndKey = (blockPath, value) => (obj) => {
 }
 
 const useChartConfig = ({ table, setTable }) => {
+  const [initialConfig, setInitialConfig] = useState(defaultConfig)
```
Comment (SteRiccio) [inline webapp/views/App/views/Data/Charts/state/hooks/useChartConfig.js:48]: is initialConfig used in the code?

**Comment by persas** [comment]
> 

Done :)

File: webapp/views/App/views/Data/Charts/components/Panel/components/BlocksBuilder/blocks/Metric/Metric.js [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+    if (!metric) {
+      setDraftMetric({ key: uuidv4() })
+    }
+  }, [])
+
+  console.log('draftValues', draftValues, Object.values(draftValues).length === order.length)
+  return (
+    <div className="custom-option-modal-container">
+      <div className="custom-option-modal-header">
+        <p>Add a new metric aggregation.</p>
```
[the remark this answers is not in the document]
Comment (persas) [reply webapp/views/App/views/Data/Charts/components/Panel/components/BlocksBuilder/blocks/Metric/Metric.js:51]: I think we can leave that for next PRs. I need to think about it better.

**Review by ramirobg94 — APPROVED** [review]

**Pull request merged**